### PR TITLE
Add doxygen and iniparser packages

### PR DESCRIPTION
**Pull request by L38s**
# Purpose of pull request

This PR adds following packages:

- doxygen
- iniparser

iniparser depends on doxygen.

# Test 
### doxygen
Run following command from console
```
root@qemuarm:~/# mkdir test
root@qemuarm:~/# cd test
root@qemuarm:~/test# doxygen -g


Configuration file `Doxyfile' created.

Now edit the configuration file and enter

  doxygen Doxyfile

to generate the documentation for your project

root@qemuarm:~/test# doxygen
Searching for include files...
Searching for example files...
Searching for images...
Searching for dot files...
Searching for msc files...
Searching for dia files...
Searching for files to exclude
Searching INPUT for files to process...
Searching for files in directory /home/root/test
Reading and parsing tag files
Parsing files
Building group list...
Building directory list...
Building namespace list...
Building file list...
Building class list...
Associating documentation with classes...
Computing nesting relations for classes...
Building example list...
Searching for enumerations...
Searching for documented typedefs...
Searching for members imported via using declarations...
Searching for included using directives...
Searching for documented variables...
Building interface member list...
Building member list...
Searching for friends...
Searching for documented defines...
Computing class inheritance relations...
Computing class usage relations...
Flushing cached template relations that have become invalid...
Computing class relations...
Add enum values to enums...
Searching for member function documentation...
Creating members for template instances...
Building page list...
Search for main page...
Computing page relations...
Determining the scope of groups...
Sorting lists...
Freeing entry tree
Determining which enums are documented
Computing member relations...
Building full member lists recursively...
Adding members to member groups.
Computing member references...
Inheriting documentation...
Generating disk names...
Adding source references...
Adding xrefitems...
Sorting member lists...
Computing dependencies between directories...
Generating citations page...
Counting data structures...
Resolving user defined references...
Finding anchors and sections in the documentation...
Transferring function references...
Combining using relations...
Adding members to index pages...
Generating style sheet...
Generating search indices...
Generating example documentation...
Generating file sources...
Generating file documentation...
Generating page documentation...
Generating group documentation...
Generating class documentation...
Generating namespace index...
Generating graph info page...
Generating directory documentation...
Generating index page...
Generating page index...
Generating module index...
Generating namespace index...
Generating namespace member index...
Generating annotated compound index...
Generating alphabetical compound index...
Generating hierarchical class index...
Generating member index...
Generating file index...
Generating file member index...
Generating example index...
finalizing index lists...
writing tag file...
lookup cache used 0/65536 hits=0 misses=0
finished...
```

### iniparser

Since iniprser does not work on the command line, you will need a program to test it.
Go to https://salsa.debian.org/carnil/iniparser
Build and run the program in the example directory to check it.
Note that the test directory requires a make environment for testing.

You need the following configs in your local.conf to make the example (Dropbear for copying the example source).

```
EXTRA_IMAGE_FEATURES_append = " tools-sdk"
EXTRA_IMAGE_INSTALL_append = " iniparser iniparser-dev dropbear"
```

There are two programs in example: "iniexample" and "parse".

When you run "iniexample", a file called "example.ini" will be created in the root directory, so it is easy to use that file as an argument for the "parse" program.

~~~
root@qemuarm:~# iniexample

root@qemuarm:~# ls
example.ini

root@qemuarm:~# parse example.ini
~~~

### doxygen
The analysis process is performed even if there is no file.
```
root@qemuarm:~/test#  ls
Doxyfile  html      latex
#
```

### iniparser

If the process is successful, you will get the parsed data.

~~~
root@qemuarm:~# iniexample
[pizza]=UNDEF
[pizza:ham]=[yes]
[pizza:mushrooms]=[TRUE]
[pizza:capres]=[0]
[pizza:cheese]=[Non]
[wine]=UNDEF
[wine:grape]=[Cabernet Sauvignon]
[wine:year]=[1989]
[wine:country]=[Spain]
[wine:alcohol]=[12.5]
Pizza:
Ham:       [1]
Mushrooms: [1]
Capres:    [0]
Cheese:    [0]
Wine:
Grape:     [Cabernet Sauvignon]
Year:      [1989]
Country:   [Spain]
Alcohol:   [12.5]
root@qemuarm:~# parse example.ini
[pizza]=UNDEF
[pizza:ham]=[yes]
[pizza:mushrooms]=[TRUE]
[pizza:capres]=[0]
[pizza:cheese]=[Non]
[wine]=UNDEF
[wine:grape]=[Cabernet Sauvignon]
[wine:year]=[1989]
[wine:country]=[Spain]
[wine:alcohol]=[12.5]
~~~

### CODE REVIEW

File: recipes-debian/doxygen/doxygen/0001-build-don-t-look-for-Iconv.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+     find_package(SQLite3 REQUIRED)
+ endif()
+ 
+-find_package(Iconv REQUIRED)
+-include_directories(${ICONV_INCLUDE_DIR})
+-
+-
+ #set(DOXYDOCS ${CMAKE_SOURCE_DIR}/doc CACHE INTERNAL "Path to doxygen docs")
+ set(DOXYDOCS ${PROJECT_BINARY_DIR}/doc)
+ set(ENV{DOXYGEN_DOCDIR} ${DOXYDOCS})
```
Comment (hiraku-wfs) [inline recipes-debian/doxygen/doxygen/0001-build-don-t-look-for-Iconv.patch:31]: Please use "git format-patch" to create a patch.
I created it this time for you.
https://gist.github.com/hiraku-wfs/80506f70fd06d54d3fad9f80c9ee0d38

You can find corresponding git repository from Debian's "Developer Information".

1. https://packages.debian.org/source/buster/doxygen
2. https://tracker.debian.org/pkg/doxygen
3. https://salsa.debian.org/debian/doxygen/-/tree/debian/buster <--

File: recipes-debian/doxygen/doxygen_debian.bb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+LICENSE = "GPL-2.0"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=b234ee4d69f5fce4486a80fdaf4a4263"
+
+DEPENDS = "flex-native bison-native"
+
+inherit debian-package
+
+SRC_URI += "file://0001-build-don-t-look-for-Iconv.patch"
+
+inherit cmake python3native
```
Comment (hiraku-wfs) [inline recipes-debian/doxygen/doxygen_debian.bb]: Please remove unnecessary space at the end of the line.

Comment (L38s) [reply recipes-debian/doxygen/doxygen_debian.bb:17]: fixed.

**Comment by hiraku-wfs** [comment]
@L38s Thank you for your contribution!

**Pull request merged**